src/ui/main_window.py

```
@@ -228,39 +245,107 @@ def _auto_load_experiment_data(self) -> None:
-
-                        QTimer.singleShot(200, load_roi_and_plot)
+                        if self.experiment.roi:
+                            loading_dialog.update_status("Loading ROI...", "Restoring ROI selection")
+                            QApplication.processEvents()
+                            
+                            roi_data = self.experiment.roi
+                            
+                            # Convert dict to ROI object
+                            try:
+                                roi = ROI.from_dict(roi_data)
+                            except Exception:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)